src/app/projects/new-project/new-project.component.spec.ts

```diff
@@ -0,0 +1,103 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { NewProjectComponent } from './new-project.component';
+import { MatBottomSheetRef } from "@angular/material";
+import { ProjectsService } from "../projects.service";
+import { Project } from "../project";
+import { Observable, of } from "rxjs";
+import { BrowserAnimationsModule } from "@angular/platform-browser/animations";
+import { LayoutModule } from "@angular/cdk/layout";
+import { FlexLayoutModule } from "@angular/flex-layout";
+import { ReactiveFormsModule } from "@angular/forms";
+import { AppMaterialModule } from "../../app-material.module";
+import { GlobalErrorHandlerService } from "../../error/global-error-handler.service";
+
+const bottomSheetStub: Partial<MatBottomSheetRef> = {
+  dismiss(result?: any): void { }
+};
+
+const projectsServiceStub: Partial<ProjectsService> = {
+  createProject(project: Project):
+    Observable<Project> { return null; }
+};
+
+describe('NewProjectComponent', () => {
+  let component: NewProjectComponent;
+  let fixture: ComponentFixture<NewProjectComponent>;
+
+  beforeEach(async(() => {
+    spyOn(projectsServiceStub, 'createProject')
+      .and.returnValue(of<Project>(
+      {
+        id: 'e6b808eb-b6bd-447d-8dce-3e0d66b17759',
+        name: 'name',
+        customer: 'customer',
+        industrySector: 'industrySector',
+        description: 'description',
+        creationDate: new Date(),
+        lastModifiedDate: new Date()
+      }));
+    TestBed.configureTestingModule({
+      imports: [
+        BrowserAnimationsModule,
+        LayoutModule,
+        FlexLayoutModule,
+        ReactiveFormsModule,
+        AppMaterialModule
+      ],
+      declarations: [ NewProjectComponent ],
+      providers: [
+        GlobalErrorHandlerService,
+        { provide: ProjectsService, useValue: projectsServiceStub },
+        { provide: MatBottomSheetRef, useValue: bottomSheetStub }
```

Comment:
I would suggest to use here `jasmine.createSpyObj('matBottomSheetRef', ['dismiss'] )` instead of bottomSheetStub.
The reasons for that are usage of a special jasmine-testing API and reducing of number of lines of source code.

Reply:
done